lib/common/output.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2019 Chris Lumens <clumens@redhat.com>
+ *
+ * This source code is licensed under the GNU General Public License version 2
+ * or later (GPLv2+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm/common/output.h>
+#include <crm/common/util.h>
+#include <crm/common/xml.h>
+#include <crm/common/internal.h>
+#include <pacemaker-internal.h>
+#include <libxml/tree.h>
+
+static GHashTable *formatters = NULL;
+
+int
+pcmk__register_base_formats() {
+    if (pcmk__register_format(&pcmk_text_formatter) != 0) {
+        return 1;
+    }
+
+    if (pcmk__register_format(&pcmk_xml_formatter) != 0) {
+        return 2;
+    }
```

Comment:
My plan here was a different error code for each formatter, so I might later be able to tell which had failed.  Perhaps it'd be enough to just fprintf something to the screen, though.

Reply:
I'd rather avoid the output functions emitting output on their own. I'm thinking pcmk__register_format() can CRM_ASSERT() its error conditions and return void, since there's really no way for it to fail unless there's a bug.